hosts/galileo/ssi/__init__.py

```diff
@@ -0,0 +1,334 @@
+    """Initialize key information about the SSI instrument.
+
+    Must be called first. After the first call, later calls to this function
+    are ignored.
+
+    Input:
+        ck,spk      'predicted', 'reconstructed', or 'none', depending on which
+                    kernels are to be used. Defaults are 'reconstructed'. Use
+                    'none' if the kernels are to be managed manually.
+        planets     A list of planets to pass to define_solar_system. None or
+                    0 means all.
+        asof        Only use SPICE kernels that existed before this date; None
+                    to ignore.
+        gapfill     True to include gapfill CKs. False otherwise.
+        mst_pck     True to include MST PCKs, which update the rotation models
+                    for some of the small moons.
+        irregulars  True to include the irregular satellites;
+                    False otherwise.
+    """
+    SSI.initialize(ck=ck, planets=planets, asof=asof,
+                   spk=spk, gapfill=gapfill,
+                   mst_pck=mst_pck, irregulars=irregulars)
+
+
+#===============================================================================
+class Metadata(object):
+
+    #===========================================================================
+    def __init__(self, label):
+        """Use the label to assemble the image metadata.
+
+        Input:
+            label           The label dictionary.
+
+        Attributes:
+            nlines          A Numpy array containing the data in axis order
+                            (line, sample).
+            nsamples        The time sampling array in (line, sample) axis
+                            order, or None if no time backplane is found in
+                            the file.
+            nframelets
+
+        """
+
+        # Image dimensions
+        self.nlines = label['IMAGE']['LINES']
+        self.nsamples = label['IMAGE']['LINE_SAMPLES']
+
+        # Exposure time
+        exposure_ms = label['EXPOSURE_DURATION']
+        self.exposure = exposure_ms/1000.
+
+        # Filters
+        self.filter = label['FILTER_NAME']
+
+        #TODO: determine whether IMAGE_TIME is the start time or the mid time..
+        self.tstart = julian.tdb_from_tai(
+                        julian.tai_from_iso(label['IMAGE_TIME']))
+        self.tstop = self.tstart + self.exposure
+
+        # Target
+        self.target = label['TARGET_NAME']
+
+        # Telemetry mode
+        self.mode = label['TELEMETRY_FORMAT_ID']
+
+        # Window
+        if 'CUT_OUT_WINDOW' in label:
+            self.window = label['CUT_OUT_WINDOW']
+        else:
+            self.window = None
+
+    #===========================================================================
+    def trim(self, data, full_fov=False):
+        """Trim image to label window
+
+        Input:
+            full_fov        If True, the image is not trimmed.
+
+        Attributes:
```

Comment:
I don't think the Attributes part of this comment is relevant

Reply:
Agreed, removed.